sinks/influxdb/driver.go

```diff
@@ -28,56 +28,65 @@ import (
 	influxdb "github.com/influxdb/influxdb/client"
 	"k8s.io/heapster/extpoints"
 	sink_api "k8s.io/heapster/sinks/api"
-	"k8s.io/heapster/util"
+	"k8s.io/heapster/version"
 	kube_api "k8s.io/kubernetes/pkg/api"
 )
 
-type influxDBClient interface {
-	WriteSeriesWithTimePrecision(series []*influxdb.Series, timePrecision influxdb.TimePrecision) error
-	GetDatabaseList() ([]map[string]interface{}, error)
-	CreateDatabase(name string) error
-	DisableCompression()
+type influxdbClient interface {
+	Write(influxdb.BatchPoints) (*influxdb.Response, error)
+	Query(influxdb.Query) (*influxdb.Response, error)
+	Ping() (time.Duration, string, error)
 }
 
 type influxdbSink struct {
-	client    influxDBClient
-	stateLock sync.RWMutex
+	client influxdbClient
+	sync.RWMutex
 	// TODO(rjnagal): switch to atomic if writeFailures is the only protected data.
 	writeFailures int // guarded by stateLock
-	seqNum        metricSequenceNum
 	c             config
+	dbExists      bool
 }
 
 type config struct {
-	user         string
-	password     string
-	host         string
-	dbName       string
-	avoidColumns bool
+	user     string
+	password string
+	host     string
+	dbName   string
+	secure   bool
 }
 
 const (
-	eventsSeriesName = "log/events"
+	eventMeasurementName = "log/events"
 	// Attempt database creation maxRetries times before quitting.
 	maxRetries = 20
 	// Sleep for waitDuration between database creation retries.
 	waitDuration = 30 * time.Second
+	// Value Field name
+	valueField = "value"
+	// Event special tags
+	eventUID        = "uid"
+	dbNotFoundError = "database not found"
 )
 
-func (self *influxdbSink) Register(metrics []sink_api.MetricDescriptor) error {
-	// Create tags once influxDB v0.9.0 is released.
+var (
+	eventPodID   = sink_api.LabelPodId.Key
+	eventPodName = sink_api.LabelPodName.Key
+	eventHost    = sink_api.LabelHostname.Key
+)
+
+func (sink *influxdbSink) Register(metrics []sink_api.MetricDescriptor) error {
+	if err := sink.createDatabase(); err != nil {
+		glog.Error(err)
+	}
+
 	return nil
```

Comment:
Now that Register does create a database, shouldn't Unregister do something too?